### ISSUE TITLE
[FEATURE] Add Refund Sync route in compatibiltiy

### ISSUE THREAD

**Issue by Abhicodes-crypto** [juspay/hyperswitch#920]
### Feature Description

There is a refund sync route https://github.com/juspay/hyperswitch/blob/573a4d384ee6a9d72648ab537804799a3993e1e8/crates/router/src/routes/app.rs#L184 

A similar route needs to be added in the compatibility layer in this file - https://github.com/juspay/hyperswitch/blob/main/crates/router/src/compatibility/stripe/refunds.rs 

### Possible Implementation

refer similar route of payments as an example. - https://github.com/juspay/hyperswitch/blob/573a4d384ee6a9d72648ab537804799a3993e1e8/crates/router/src/compatibility/stripe/app.rs#L12 

### Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### Are you willing to submit a PR?

None

**Comment by shubhanshu-mani** [comment]
Hi I would like to pick this up , though I need some additional details on this. 

1. How is refund sync different from refund-retrieve https://github.com/juspay/hyperswitch/blob/573a4d384ee6a9d72648ab537804799a3993e1e8/crates/router/src/compatibility/stripe/app.rs#L41
2. Do you want something like refund_retrieve_with_gateway_creds

**Comment by Abhicodes-crypto** [comment]
> How is refund sync different from refund-retrieve

It calls the same function of refund sync. Only difference is that in the request body the gateway creds are passed.

> Do you want something like refund_retrieve_with_gateway_creds

Yes.

**Comment by shubhanshu-mani** [comment]
Hey do I need to fork this repo and create a PR from that? Or are people allowed to push branches directly

**Comment by SanchithHegde** [comment]
> Hey do I need to fork this repo and create a PR from that? Or are people allowed to push branches directly

Yes @shubhanshu-mani, you can fork the repository and open a PR from your fork. Users who aren't members of the `juspay` GitHub organization aren't allowed to create branches on our repository.

**Comment by shubhanshu-mani** [comment]
@SanchithHegde sure I have created the PR from fork . BTW my account is part of juspay organisation don't know why it was allowing me to push branch directly

**Comment by SanchithHegde** [comment]
> BTW my account is part of juspay organisation don't know why it was not allowing me to push branch directly

We have a further restriction that only users within specific teams in the GitHub organization have write access to the repository.
